src/domain_knowledge.jl

```diff
@@ -31,6 +31,11 @@ This is within the policy when a RolloutEstimator is passed to an AbstractMCTSSo
 mutable struct SolvedRolloutEstimator{P<:Policy, RNG<:AbstractRNG}
     policy::P
     rng::RNG
+    depth::Union{Int, Nothing}
+
+    function SolvedRolloutEstimator(policy, rng, depth::Union{Int, Nothing}=nothing)
```

Comment:
By default here, rollouts will run until a terminal state is hit. Provided that there are plenty POMDPs/MDPs without terminal states (tiger, crying baby, etc.) these defaults will fail silently with the rollout running in an infinite loop. Even if we do throw one warning deterring users from falling into this infinite loop trap, it would be an unnecessary nuisance.

I would recommend setting some default maximum depth or default epsilon that prevents this.

Also, given that the end-user doesn't usually directly interact with `SolvedRolloutEstimator`, this constructor may be unnecessary.

Reply:
Thanks for having a look. I agree that cyclic/infinite horizon (PO)MDPs should be handled gracefully. 

I think the default rollout estimator with finite depth and eps<1 should be fine, passing these values to the `RolloutSimulator`. There will be a slight inconsistency since the defaults for `RolloutSimulator` are `nothing` for both values. But that's better than having the infinite loops I think.

I have a question though, what do you think about removing the `depth` value from the call to `estimate_value` in this PR? IMO its an improvement, but it is a significant change since it will be impossible to replicate the old rollout behavior (with `depth` parameter bounding _combined_ depth of tree _and_ rollout - with different rollouts being of different depth as a result). Do you know if the old behavior is important to anyone?

It is possible to keep the `depth` parameter in the calls to `estimate_value` and have the new `RolloutEstimator` handling the new way of rollouts, and then add a `LegacyRolloutEstimator` replicating the old behavior, but I think it would be better to remove this complexity if it is not needed.

Reply:
I absolutely believe this to be a worthwhile pursuit. I also totally forgot that `RolloutSimulator` has `nothing` as both defaults, however because people tend to use a random rollout simulator as the default value estimator (MCTS, POMCPOW, etc.) the depth is hidden behind a second layer of abstraction.

It certainly wouldn't be impossible to maintain the legacy functionality, however I don't see the need to do so given that the old approach has had some noted problems and doesn't seem to offer an strict benefits relative to this newer approach. In my opinion you're right in that it would be better to minimize complexity and change over to constant depth rollouts while dropping this old functionality.